qutip/tests/test_qubitcircuit.py

```diff
@@ -148,6 +148,24 @@ def testSNOTdecompose(self):
         U2 = gates.gate_sequence_product(qc2.propagators())
         assert _op_dist(U1, U2) < 1e-12
 
+    def testFREDKINdecompose(self):
+        """
+        FREDKIN to rotation and CNOT: compare unitary matrix for FREDKIN and product of
+        resolved matrices in terms of rotation gates and CNOT.
+        """
+        qc1 = QubitCircuit(3)
+        print("OK")
+        qc1.add_gate("FREDKIN", targets=[0, 1], controls=[2])
+        print("OK")
+        U1 = gates.gate_sequence_product(qc1.propagators())
+        print("OK")
+        qc2 = qc1.resolve_gates()
+        print("OK")
+        U2 = gates.gate_sequence_product(qc2.propagators())
+        print("OK")
```

Comment:
There is no need to print ok here. If something goes wrong, pytest will locate which line the error comes from and summarize it in the result.